conwayste/src/ui/layer.rs

```diff
@@ -31,137 +31,251 @@ use super::{
 
 use crate::constants::colors::*;
 
-pub struct Layer {
-    pub id: WidgetID,
-    pub widgets: Vec<Box<dyn Widget>>,
+pub struct Layering {
     pub with_transparency: bool,
-    pub focused_widget: Option<WidgetID>,
+    pub widget_list: Vec<Box<dyn Widget>>,
+    focused_ids: Vec<Option<WidgetID>>,
+    id_cache: Vec<WidgetID>
 }
 
-/// A container of one or more widgets or panes
-impl Layer {
-    /// Specify the unique widget identifer for the layer
-    pub fn new(widget_id: WidgetID) -> Self {
-        Layer {
-            id: widget_id,
-            widgets: vec![],
+/// A container of one or more widgets or panes ordered by virtual layers.
```

Comment:
Because we're using a number to describe the z-order, there isn't a grouping/collection for widgets at a depth (unlike before). Thus I described it as virtual here. But I think I'll just replace it with something else. Let me know if the later commit to address this is insufficient.

Reply:
Can you explain how the optional transparency works? It's still... unclear... :sunglasses: . Also this would be a good place to say if the widget with the smaller or larger z-index is on top.

Reply:
Yes, thanks for this! Found a discrepancy with how I *thought* it worked versus what the code was doing. Updated in a3537c4.